gitcache/main.go

```diff
@@ -4,139 +4,191 @@ import (
 	"fmt"
 	"io/ioutil"
 	"log"
-	"net/url"
 	"os"
 	"path"
-	"strings"
 	"time"
 
 	"github.com/go-git/go-git/v5"
 	"github.com/mholt/archiver"
+	"github.com/pkg/errors"
+	"go.uber.org/zap"
 )
 
 func main() {
-	blob := os.Getenv("BLOB_URL")
+	const gittar string = "gitfolder.tar.gz"
+	var gitRepo *git.Repository
 
-	if blob == "" {
-		log.Panic("BLOB_URL env is not set.")
-	}
+	logLevel := zap.LevelFlag("verbosity", zap.InfoLevel, "override the default log level")
 
-	repo := &RepoURL{}
-	err := repo.Set(os.Args[1])
+	cfg := zap.NewProductionConfig()
+	cfg.Level.SetLevel(*logLevel)
+	logger, err := cfg.Build()
 	if err != nil {
 		log.Panic(err)
 	}
+	//nolint
+	defer logger.Sync() // flushes buffer, if any
+	sugar := logger.Sugar()
+	blob := os.Getenv("BLOB_URL")
+	sugar.Info("BLOB_URL ", blob)
 
-	cwd, err := os.Getwd()
-	if err != nil {
-		log.Panic(err)
+	if blob == "" {
+		sugar.Panic("BLOB_URL env is not set.")
+	}
+
+	// opening the blob
+	bucket, e := New(blob)
```

Comment:
This New() call here is also a bit confusing, being in the main package. Let's rename this to NewBucket or something similar.

Reply:
Agreed. Thanks